src/javascript/app/base/header.js

```diff
@@ -624,8 +626,10 @@ const Header = (() => {
         Login.redirectToLogin();
     };
 
-    const logoutOnClick = () => {
-        Client.sendLogoutRequest();
+    const logoutOnClick = async () => {
+        const onLogoutWithOauth = await AuthClient.getLogoutHandler(Client.sendLogoutRequest);
+
```

Comment:
This will wrap the logout call `Client.sendLogoutRequest` with our own logout iframe, which is to inform Hydra that the user is logging out and the session should be cleared on Hydra's side. Once this is done, it will call the passed-in logout handler `Client.sendLogoutRequest`.
If Hydra authentication is not enabled, the logout handler `Client.sendLogoutRequest` will just be called instead.

Reply:
this explanation is great, though you can also add it as comment too